substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jfr/JfrChunkWriter.java

```diff
@@ -61,27 +62,33 @@
 public final class JfrChunkWriter implements JfrUnlockedChunkWriter {
     public static final byte[] FILE_MAGIC = {'F', 'L', 'R', '\0'};
     public static final short JFR_VERSION_MAJOR = 2;
-    public static final short JFR_VERSION_MINOR = 0;
+    public static final short JFR_VERSION_MINOR = 1;
     private static final int CHUNK_SIZE_OFFSET = 8;
-
-    public static final long METADATA_TYPE_ID = 0;
-    public static final long CONSTANT_POOL_TYPE_ID = 1;
-
+    private static final int FILE_STATE_OFFSET = 64;
+    private static final byte COMPLETE = 0;
+    private static final short FLAG_COMPRESSED_INTS = 0b01;
+    private static final short FLAG_CHUNK_FINAL = 0b10;
     private final JfrGlobalMemory globalMemory;
-    private final ReentrantLock lock;
+    private final VMMutex lock;
```

Comment:
The only reason was so that the "write" methods (`writeBoolean`, `writeByte` etc. ) in JfrChunkWriter could be made uninterruptible so that they could be called from the `JfrConstantPool.write` methods (which are now uninterruptible).

This is really only necessary for the assertions which call interruptible `ReentrantLock` methods.